Il2CppInterop.Runtime/IL2CPP.cs

```diff
@@ -236,7 +236,7 @@ public static IntPtr GetIl2CppNestedType(IntPtr enclosingType, string nestedType
         }
 
         while ((nestedTypePtr = il2cpp_class_get_nested_types(enclosingType, ref iter)) != IntPtr.Zero)
-            if (Marshal.PtrToStringAnsi(il2cpp_class_get_name(nestedTypePtr)) == nestedTypeName)
+            if (Marshal.PtrToStringUTF8(il2cpp_class_get_name(nestedTypePtr)) == nestedTypeName)
```

Comment:
I find it difficult to believe this is the only place. Strings should be either all Ansi or all UTF8.

Reply:
My old PR loaded System.String everywhere.
https://github.com/BepInEx/Il2CppInterop/pull/110/commits/abdebd58730245b67d349434b40400b6a90fbb01

Reply:
thanks
I've now had time to test the patch and I've not found any issue so far.